### PR TITLE
docs(torghut): add visual transformer math notebook

### PR DESCRIPTION
**Pull request by gregkonush**
## Summary

- Add `40-transformer-math-from-first-principles.ipynb`, a 30-section decoder-only transformer curriculum that progresses from embeddings and matrix shapes through Q/K/V, scaled dot-product attention, causal masking, softmax, multi-head attention, RoPE, residuals, normalization, MLP/SwiGLU, logits, cross-entropy, gradients, and end-to-end attention arithmetic.
- Pair every numbered mathematical section with an executable NumPy/Plotly geometric visualization, including vector arrows, matrix rectangles, lower-triangular masks, probability simplices, convex hulls, rotation circles, residual parallelograms, quadratic scaling squares, linear KV-cache rectangles, and FlashAttention tiles.
- Include the output-free notebook and its README in the Torghut Jupyter image, seed them into the editable workspace on first launch, and include the new image inputs in notebook CI/build/release change detection.
- Add notebook execution/content contracts that require 30 progressively ordered sections, a geometric code cell after every section, unique cell IDs, no committed outputs, complete top-to-bottom execution, and runtime image/entrypoint coverage.

## Related Issues

None.

## Testing

- `services/torghut/.venv/bin/python -m pytest services/torghut/tests/notebook_data -q` — 49 passed.
- `uv run --frozen pyright --project pyrightconfig.json` — 0 errors.
- `uv run --frozen pyright --project pyrightconfig.alpha.json` — 0 errors.
- `uv run --frozen pyright --project pyrightconfig.scripts.json` — 0 errors.
- `bash -n services/torghut/scripts/start_torghut_notebook.sh` — passed.
- Parsed the three modified GitHub Actions workflows with PyYAML and validated the notebook with `nbformat`; 63 unique cells, 31 code cells, and zero committed outputs.
- `git diff --check` — passed.

## Breaking Changes

None.

## Checklist

- [x] Testing section documents the exact validation performed.
- [x] Screenshots section removed; the notebook contains executable inline visualizations.
- [x] Documentation and notebook runtime packaging are updated.

### CODE REVIEW

File: services/torghut/notebooks/40-transformer-math-from-first-principles.ipynb [hunk 1]
```diff
@@ -0,0 +1,1685 @@
...
+   "cell_type": "markdown",
+   "id": "c02a294b10d3",
+   "metadata": {},
+   "source": [
+    "## 1. A token is a point in an embedding space\n",
+    "\n",
+    "For vocabulary size \\(V\\) and model width \\(d\\), the embedding table is\n",
+    "\n",
+    "\\[\n",
+    "E \\in \\mathbb{R}^{V \times d}.\n",
```
[the remark this answers is not in the document]
Comment (gregkonush) [reply services/torghut/notebooks/40-transformer-math-from-first-principles.ipynb]: Fixed in 3760f4d75. I replaced all parsed tab control characters with literal LaTeX `\times` and `\top` commands across the notebook (5 and 12 occurrences respectively). Added a regression assertion that scans every markdown cell and fails on any control character other than newline/carriage return. Validation: focused notebook tests 2 passed, full notebook suite 49 passed, and all three Pyright profiles report 0 errors.

[the remark this answers is not in the document]
Comment (gregkonush) [reply services/torghut/notebooks/40-transformer-math-from-first-principles.ipynb]: Fixed in 3760f4d75. I rewrote the affected notebook markdown through nbformat so `\times` and `\top` survive JSON parsing as LaTeX commands, verified that all markdown cells contain no control characters, and added a regression assertion in `test_transformer_math_notebook.py`. The notebook executes top to bottom and all 49 notebook tests pass.

**Comment by gregkonush** [comment]
@codex review
